### PR TITLE
feat(pre-commit): Pre commit parametrisation - cont

### PR DESCRIPTION
**Pull request by horiagunica**
## Description

- Parametrised pre-commit workflow to run tests based on workflow inputs from source repository
  - Two new inputs are defined :
    -  pre-commit-hooks
        - Mandatory : yes
        - Default values : all the checks included
    -  pre-commit-files
        - Mandatory : no
        - No default value

- Removed checkov workflow entirely (now runs based on pre-commit input while lusing local .pre-commit.yml in local source repository)
- Added pre-commit.yml file to this workflow CI repository which includes only checkov checks .
- Added pre-commit workflow call to actions_release_ci workflow in order to trigger checkov at release .

## Motivation and Context

- Centralised CI workflows enhancement and pre-commit parametrisation .

## How Has This Been Tested?

- Used a separate repo that called sourced workflows CI in current "pre-commit-parametrisation" development branch.
  - Tested checkov fail/pass
  - Tested terraform fmt fail/pass
  - Tested terraform docs fail/pass

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes if appropriate.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/actions_release_ci.yml [hunk 1]
```diff
@@ -10,6 +10,39 @@ on:
...
+    runs-on: ubuntu-latest
+    outputs:
+      changed_files: ${{ steps.format.outputs.files_diff }}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: get diff with base branch
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/actions_release_ci.yml:25]: as discussed, this can even be the same CI, just doing different things depending on the trigger

**Review by FoSix — CHANGES_REQUESTED** [review]

**Comment by FoSix** [comment]
and one more bug, you did not change the release CI :) it still refers to the original `_checkov.yml` workflow :) 

see [here](https://github.com/PaloAltoNetworks/terraform-modules-vmseries-ci-workflows/blob/a76477bc5c8415d8553334527d780472ab704a14/.github/workflows/release_ci.yml#L71)

```
[Invalid workflow file: .github/workflows/release_ci.yml#L21](https://github.com/PaloAltoNetworks/vm-series-gh-actions/actions/runs/4191953420/workflow)
error parsing called workflow
".github/workflows/release_ci.yml"
-> "PaloAltoNetworks/terraform-modules-vmseries-ci-workflows/.github/workflows/release_ci.yml@pre-commit-parametrization" (source branch with sha:8db5c40195168451ca1624351a08f8ca025b4b9e)
--> "./.github/workflows/_checkov.yml"
: failed to fetch workflow: workflow was not found.
```

File: README.md [hunk 2]
```diff
@@ -71,11 +71,7 @@ flowchart
     subgraph P [Pre-Commit Sub-Workflow]
         direction TB
         p_start(Init point of a sub-workflow)-->p_prereq(Install pre-commit prerequisites)
-        p_prereq-->p_pre_commit(Run Pre-Commit)
-    end
-    subgraph C [Checkov Sub-Workflow]
-        direction TB
-        c_start(Init point of a sub-workflow)-->c_checkov(Run Checkov)
+        p_prereq-->p_pre_commit(Run validation strategy: matrix)
```
[the remark this answers is not in the document]
Comment (horiagunica) [reply README.md:74]: Modified to " p_prereq-->p_pre_commit(Execute pre-commit hooks based on input parametrization) "

**Review by FoSix — APPROVED** [review]
👍🏻

**Pull request merged**